### DISCUSSION TITLE
Improving error diagnostics on Sphinx build

### DISCUSSION THREAD

**Discussion by bertiewooster** [sympy/sympy#23680] [Ideas]
Is there a way to set up the Sphinx documentation build so it will give more information when it fails, namely which file and ideally line number didn't parse correctly? At present, all it gives is:

`make: *** [Makefile:62: html] Error 1`

which means you have to guess which file and line failed.

I'm using the Docker build method, if that makes any difference.

**Comment by asmeurer** [comment]
Normally the warnings would be colored red, but it looks like Circle disables color output when the log is too long https://discuss.circleci.com/t/ansi-colours-are-not-supported-in-the-new-ui-2020/35065/2. I'm going to try splitting it the install phase to see if that fixes it.

**Reply by asmeurer** [reply]
https://github.com/sympy/sympy/pull/23695

**Comment by asmeurer** [comment]
Oh I guess you're talking about building locally with Docker. Does the Docker build remove the colors? If so, we should see if that can be fixed. 

Otherwise, I don't know if we can make Sphinx print all the warnings at the end. We'll need to investigate whether that's possible. It would be useful if it could.

**Comment by asmeurer** [comment]
You can use the `-w` flag to write the warnings to a file in addition to stderr. So I think we could modify the Makefile to write the warnings to a temporary file then write those warnings to the terminal at the end.

**Comment by asmeurer** [comment]
I opened an issue to track these things https://github.com/sympy/sympy/issues/23697.

**Reply by bertiewooster** [reply]
Great, thanks! The Sphinx error reporting is a great improvement.
